python_modules/libraries/dagster-components/dagster_components/core/component_decl_builder.py

```diff
@@ -33,6 +33,45 @@ class ComponentFileModel(BaseModel):
 T = TypeVar("T", bound=BaseModel)
 
 
+def has_python_files(folder_path: Path) -> bool:
+    """Check if a folder contains any Python files excluding __init__.py.
+
+    Args:
+        folder_path (Path): Path object representing the folder to check
+    Returns:
+        bool: True if folder contains .py files (except __init__.py), False otherwise
+    """
+    if not folder_path.is_dir():
+        return False
+    # Check all files in the directory
+    for file_path in folder_path.glob("*.py"):
+        if file_path.name != "__init__.py":
```

Comment:
why exclude init.py?

Reply:
Yeah I think I wrote this but @benpankow is right here IMO